### PR TITLE
Remove redundant shims from `shell/types/rancher/index.d.ts`

### LINKED ISSUE #18548: Remove redundant shims from `shell/types/rancher/index.d.ts`
`shell/types/rancher/index.d.ts` declares shims that masks actual exports, making them invisible to the type checker. This leads to type errors in imports. For example:

```
Module '"@shell/config/query-params"' has no exported member '_EDIT'.
```

## Acceptance criteria

- [ ] `shell/types/rancher/index.d.ts` removes redundant instances of `declare module` blocks
- [ ] `yarn type-check:ci` passes
- [ ] Baseline total is reduced and the committed baseline reflects it
- [ ] `yarn test:ci` passes
- [ ] No production source files modified; small runtime-only annotation are expected to change for some edge-cases that were previously outdated because of the shims 

## Additional context

Consider reviewing shims in other components as well:

- `@shell/store/type-map`
- `@shell/config/version`
- `@shell/plugins/dashboard-store`

### PR DESCRIPTION
**Pull request by rak-phillip**
### Summary
This removes shims that were at the root of common type errors related to imports throughout Rancher Dashboard.

Fixes #18548 

### Occurred changes and/or fixed issues

Removed:
- `declare module '@shell/store/type-map'`
- `declare module '@shell/plugins/dashboard-store'`
- `declare module '@shell/config/query-params'`

Resolved type errors in `shell/edit/workload/__tests__/Upgrading.test.ts` that arose as a result of the shim change

Updated the baseline, reducing the total errors down from 1977 to 1725

### Technical notes summary

I think this is fairly low-hanging fruit to pay down some of the debt. If I recall, I think that most of these shims were introduced to deal with Vue2 deficiencies when working with typescript.

Each of these shims appear to be safe to remove. In quite a few cases, they were inaccurate and out of date, so their removal is warranted. 

### Areas or cases that should be tested

CI should pass all gates. No new type check warnings should be raised as a result of this change.

### Areas which could experience regressions

No behavior should be altered as a result of this change. 

### Screenshot/Video

NA

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes
- [x] The PR has been reviewed in terms of Accessibility
- [x] The PR has considered, and if applicable tested with, the three Global Roles `Admin`, `Standard User` and `User Base`

### CODE REVIEW

**Comment by rak-phillip** [comment]
@richard-cox, you raised some concerns about the shims Rancher Shell bloat: https://github.com/rancher/dashboard/issues/18548#issuecomment-5127298251

I'm not 100% sure what we would want to validate to ensure that we don't regress with this change.

**Comment by richard-cox** [comment]
@rak-phillip might want to double check with @nwmac but i think it was just analysing the `@rancher/shell` package and compare what it ships with and without the change.

**Comment by rak-phillip** [comment]
@richard-cox @nwmac it looks like the surface area does grow from 205 to 216 modules with this change. From what I understand, these additions would have always been visible to extensions in one way or another and that the shims were an outdated, manually maintained representation of these modules. 

These shims are at the root of quite a few type errors. An alternative path might be to try and update and maintain them, but there's a high probability that they will become outdated in the future as we continue modify shell and Dashboard. I think that removing the shims makes sense and helps our tooling to work for us.

File: shell/plugins/dashboard-store/index.js [hunk 1]
```diff
@@ -25,7 +25,9 @@ export const coreStoreState = (namespace, baseUrl, isClusterStore) => ({
   },
   types:       {},
   savedCounts: {}, // Saved counts for resource types (from paginated API called where marked)
-  $ctx:        markRaw({}),
+  // Annotated so `typegen.sh` can emit a declaration for this module; without it
+  // declaration emit fails on the private `RawSymbol` type behind `markRaw`.
+  $ctx:        /** @type {any} */ (markRaw({})),
```
[the remark this answers is not in the document]
Comment (rak-phillip) [reply shell/plugins/dashboard-store/index.js:30]: > If that criterion is still required, consider moving the typegen workaround out of the production module (e.g., a typegen-only TS/JSDoc override or a dedicated .d.ts used only during declaration emit) 

I don't agree with this. The doc overrides/dedicated `.d.ts` will reintroduce the same issue that we're resolving with the shim removal: allowing the tooling to declare the types instead of having to manually maintain something that will drift as the underlying source code changes. 

I'm opting to update the acceptance criteria so that we can move forward with this one.

File: shell/scripts/typegen.sh [hunk 2]
```diff
@@ -29,9 +29,11 @@ ${BASE_DIR}/node_modules/.bin/tsc ${SHELL_DIR}/store/features.js --declaration -
 ${BASE_DIR}/node_modules/.bin/tsc ${SHELL_DIR}/store/plugins.js --declaration --allowJs --emitDeclarationOnly --outDir ${SHELL_DIR}/tmp/store > /dev/null
 ${BASE_DIR}/node_modules/.bin/tsc ${SHELL_DIR}/store/prefs.js --declaration --allowJs --emitDeclarationOnly --outDir ${SHELL_DIR}/tmp/store > /dev/null
 ${BASE_DIR}/node_modules/.bin/tsc ${SHELL_DIR}/store/store-types.js --declaration --allowJs --emitDeclarationOnly --outDir ${SHELL_DIR}/tmp/store > /dev/null
+${BASE_DIR}/node_modules/.bin/tsc ${SHELL_DIR}/store/type-map.js --declaration --allowJs --emitDeclarationOnly --outDir ${SHELL_DIR}/tmp/store > /dev/null
 
 # # plugins
 ${BASE_DIR}/node_modules/.bin/tsc ${SHELL_DIR}/plugins/i18n.js --declaration --allowJs --emitDeclarationOnly --outDir ${SHELL_DIR}/tmp/ > /dev/null
+${BASE_DIR}/node_modules/.bin/tsc ${SHELL_DIR}/plugins/dashboard-store/index.js --declaration --allowJs --emitDeclarationOnly --outDir ${SHELL_DIR}/tmp/plugins/dashboard-store/ > /dev/null
 ${BASE_DIR}/node_modules/.bin/tsc ${SHELL_DIR}/plugins/dashboard-store/normalize.js --declaration --allowJs --emitDeclarationOnly --outDir ${SHELL_DIR}/tmp/plugins/dashboard-store/ > /dev/null
```
[the remark this answers is not in the document]
Comment (rak-phillip) [reply shell/scripts/typegen.sh:37]: This has been resolved via `45f06ee9e3d5c5446cf8b839d13fab5a24237f1a`. Although it was fairly minor, I think it makes sense to make it consistent with other imports.

**Comment by rak-phillip** [comment]
> A remaining @shell/config/version shim still masks real exports and contradicts the stated goal of removing redundant/inaccurate shims.

Removing the `@shell/config/version` shim has a higher risk associated with it than the ones represented in this PR. I want to limit the scope to the shims represented in this PR.

**Comment by aalves08** [comment]
@rak-phillip do extensions build fine with these changes? Couldn't a test like:

1. Run `./shell/scripts/typegen.sh` to generate the type definitions
2. Run `yarn link` in the `shell` package of this branch
3. On the extension side — https://github.com/rancher/supportability-review-app, branch `main` — run `yarn link "@rancher/shell"`
4. Run `yarn build-pkg supportability-review-app` to build the extension

be enough to validate it? 

Covering a few of the official extensions should suffice https://github.com/rancher/ui-plugin-charts/blob/main/manifest.json

**Comment by rak-phillip** [comment]
@aalves08 yes, extension builds fine while using yarn link with this change in place. I tested with Supportability Review, AI UI, and Ali UI.

**Review by aalves08 — CHANGES_REQUESTED** [review]
Nice one, this is a good clean-up and the diagnosis is right — those shims were stale hand-maintained fictions that were actively lying to the type checker. I pulled the branch down at `45f06ee` and went through it properly. Baseline looks honest to me: 257 removals and **zero** additions, of which 255 are `TS2305 "has no exported member"` against exactly the modules you unshimmed. The header total (1782) matches the real non-comment line count, so the baseline wasn't hand-tweaked. `yarn test:ci shell/edit/workload/__tests__/Upgrading.test.ts` passes 13/13, and there are no leftover `@shell/plugins/dashboard-store/index` imports anywhere in `shell`, `pkg`, `creators`, `cypress` or `storybook`.

I've got one thing I'd like fixed before this goes in.

---

## Required change

### The new `dashboard-store` typegen entry emits broken relative imports into the published `.d.ts`

**What's happening**

Adding `plugins/dashboard-store/index.js` to `typegen.sh` makes `tsc` emit a declaration for `coreStoreModule`, and because `index.js` re-exports things it pulls in from siblings, that declaration contains relative type references:

```ts
load:         typeof import("./mutations").load;
loadAll:      typeof import("./mutations").loadAll;
batchChanges: typeof import("./mutations").batchChanges;
loadAdd:      typeof import("./mutations").loadAdd;
forgetType:   typeof import("./mutations").forgetType;
loadSchemas:  typeof import("./actions").loadSchemas;
```

The problem is what `typegen.sh` does next. `processDir` `cat`s that file verbatim into `shell/types/shell/index.d.ts` wrapped in `declare module '@shell/plugins/dashboard-store' { … }`. Inside an ambient module block, a relative specifier resolves against **the file doing the declaring**, not against the module being declared. So `./mutations` is looked up as `@rancher/shell/types/shell/mutations`, which doesn't exist.

It's compounded by the new `rm -f …/tmp/plugins/dashboard-store/mutations.d.ts`, which means there isn't even a `declare module '@shell/plugins/dashboard-store/mutations'` block in the output to fall back on.

The reason CI is green on this is that nothing actually type-checks the generated file:

- `typegen.sh` only runs from `publish-shell.sh:107`, i.e. at publish time and inside `check-plugins-build`.
- This PR adds `shell/types/shell` to the `exclude` list in `tsconfig.type-check.json`, so `type-check` never sees it.
- The extension `pkg` template (`creators/extension/pkg/files/tsconfig.json`) sets `skipLibCheck: true`, which swallows the errors — and silently degrades those six mutations to `any`.
- The **app** template (`creators/extension/app/files/tsconfig.json`) has no `skipLibCheck`, so there they'd surface as hard errors.

So this ships in `@rancher/shell` and gets found by extension authors rather than by us.

**How to reproduce**

There's no unit test that covers this — it's a build-artifact problem — so this is a shell repro. From the repo root on the PR branch:

```bash
./shell/scripts/typegen.sh
grep -n 'import("\./' shell/types/shell/index.d.ts
```

Actual output:

```
6662:        load: typeof import("./mutations").load;
6675:        loadAll: typeof import("./mutations").loadAll;
6676:        batchChanges: typeof import("./mutations").batchChanges;
6682:        loadAdd: typeof import("./mutations").loadAdd;
6709:        forgetType: typeof import("./mutations").forgetType;
6714:        loadSchemas: typeof import("./actions").loadSchemas;
```

To confirm they're genuinely unresolvable rather than just ugly, stand up a throwaway consumer that type-checks against the generated file with `skipLibCheck` off:

```bash
mkdir -p /tmp/shimtest
grep -v '^/// <reference' shell/types/shell/index.d.ts > /tmp/shimtest/shell.d.ts

cat > /tmp/shimtest/test.ts <<'EOF'
import { coreStoreModule } from '@shell/plugins/dashboard-store';

export const a = coreStoreModule.mutations.load;
EOF

cat > /tmp/shimtest/tsconfig.json <<'EOF'
{
  "compilerOptions": {
    "target": "ES2018",
    "module": "ESNext",
    "moduleResolution": "Node",
    "strict": true,
    "noEmit": true,
    "skipLibCheck": false
  },
  "files": ["shell.d.ts", "test.ts"]
}
EOF

./node_modules/.bin/tsc -p /tmp/shimtest/tsconfig.json 2>&1 | grep "Cannot find module '\./"
```

Actual output:

```
/tmp/shimtest/shell.d.ts(6660,29): error TS2307: Cannot find module './mutations' or its corresponding type declarations.
/tmp/shimtest/shell.d.ts(6673,32): error TS2307: Cannot find module './mutations' or its corresponding type declarations.
/tmp/shimtest/shell.d.ts(6674,37): error TS2307: Cannot find module './mutations' or its corresponding type declarations.
/tmp/shimtest/shell.d.ts(6680,32): error TS2307: Cannot find module './mutations' or its corresponding type declarations.
/tmp/shimtest/shell.d.ts(6707,35): error TS2307: Cannot find module './mutations' or its corresponding type declarations.
/tmp/shimtest/shell.d.ts(6712,36): error TS2307: Cannot find module './actions' or its corresponding type declarations.
/tmp/shimtest/shell.d.ts(8147,25): error TS2307: Cannot find module './hybrid-class' or its corresponding type declarations.
/tmp/shimtest/shell.d.ts(8188,25): error TS2307: Cannot find module './hybrid-class' or its corresponding type declarations.
```

Two caveats so you can weigh this fairly. The two `./hybrid-class` lines are **pre-existing on master** (same class of bug in the `steve-class` block, static-import form) — not yours. And the harness reports 179 errors in total, but the rest are artefacts of running from `/tmp` with no `vue`/`vuex` resolvable; I only counted the `TS2307`s above, which are real regardless of environment.

I also checked whether the `rm -f` lines remove anything we used to publish, and they don't:

```bash
./node_modules/.bin/tsc shell/plugins/dashboard-store/actions.js \
  --declaration --allowJs --emitDeclarationOnly --outDir /tmp/dsprobe/master
ls /tmp/dsprobe/master
# actions.d.ts  normalize.d.ts

./node_modules/.bin/tsc shell/plugins/dashboard-store/index.js \
  --declaration --allowJs --emitDeclarationOnly --outDir /tmp/dsprobe/pr
ls /tmp/dsprobe/pr
# actions.d.ts  getters.d.ts  index.d.ts  model-loader-require.d.ts
# model-loader.d.ts  mutations.d.ts  normalize.d.ts
```

So `mutations.d.ts` and friends only appear *because of* the new `index.js` entry. Dropping them is net-neutral for existing consumers — the issue is purely that `index.d.ts` still points at one of them.

**Possible solution**

Two options.

*Minimal (what I'd pick for this PR)* — keep `mutations.d.ts`, and rewrite relative specifiers to the ambient module name as you combine. In `shell/scripts/typegen.sh`:

```diff
 # Drop transitively-emitted declarations we do not want to publish
 rm -f ${SHELL_DIR}/tmp/plugins/dashboard-store/getters.d.ts
-rm -f ${SHELL_DIR}/tmp/plugins/dashboard-store/mutations.d.ts
+# mutations.d.ts is deliberately kept: index.d.ts references it
 rm -f ${SHELL_DIR}/tmp/plugins/dashboard-store/model-loader.d.ts
 rm -f ${SHELL_DIR}/tmp/plugins/dashboard-store/model-loader-require.d.ts
```

and in `processDir`:

```diff
         echo "declare module '${module}' {" >> ${INDEX}
-        cat $entry >> ${INDEX}
+        sed 's#import("\./#import("'"${basePkg}"'/#g' $entry >> ${INDEX}
         echo -e "}" >> ${INDEX}
```

I ran this and re-checked: `grep -c 'import("\./' shell/types/shell/index.d.ts` goes to `0`, all six become `import("@shell/plugins/dashboard-store/mutations")` / `…/actions`, `declare module '@shell/plugins/dashboard-store/mutations'` is present at line 6837, and the six `TS2307`s disappear from the harness above.

*Better, but bigger* — apply the same rewrite at both `cat $entry` sites (the `${module}` one and the `${moduleWithJs}` one) and extend the pattern to static `from './…'` specifiers as well. That also cleans up the two pre-existing `./hybrid-class` references. It's the right long-term shape, but it touches output for modules this PR isn't otherwise changing, so I'd rather see it as a follow-up than bolted on here.

Either way, it'd be worth a follow-up issue to get the generated `index.d.ts` type-checked by *something* in CI, otherwise the next person to add a typegen entry hits the exact same trap.

---

## Not blocking

- The PR description says the baseline went 1977 → 1725, but the committed baseline says 2039 → 1782. Just stale after the rebase — worth updating so the numbers match what reviewers see.
- Same relative-import bug already exists on master for `./hybrid-class` in the `steve-class` block. Not introduced here; flagging it so it doesn't get lost.
- `$ctx: /** @type {any} */ (markRaw({}))` is technically a production file, but the issue's acceptance criteria explicitly allow runtime-only annotations and it's a pure type cast with no runtime effect, so I'm fine with it. `Record<string, any>` would erode a fraction less if you feel like it.
- Agreed with leaving the `@shell/config/version` shim alone — Copilot's second review is wrong to treat it as in-scope for this PR.

---

## What's good

Clearing 255 genuinely-bogus errors with zero new baseline entries is a real result, and the underlying call is correct: `DSL` was shimmed as `DSL(store, name: string)` when the real signature is `DSL(store, product, module = 'type-map')`, and `query-params` was shimmed to export only `_DETAIL` while the actual module exports `_CREATE`, `_EDIT`, `_VIEW`, `_IMPORT` and about twenty more. Those shims weren't protecting anything, they were just wrong.

Catching the `/index` resolution problem in the follow-up commit, excluding the generated directory from `type-check` so local artifacts don't pollute the run, and actually `yarn link`-testing three real extensions rather than assuming — all of that is the careful version of this change. And using `as const` on the `it.each` table instead of casting your way out of the `TS7053` was the right instinct.

**Comment by rak-phillip** [comment]
@aalves08 the latest commit should resolve the problem you detailed. 

- `> grep -n 'import("\./' shell/types/shell/index.d.ts` no longer returns a value.
- `> ./node_modules/.bin/tsc -p /tmp/shimtest/tsconfig.json 2>&1 | grep` now only returns two items related to `./hybrid-class`, which is specified as pre-existing on master. 

Can you take another look?

**Review by aalves08 — DISMISSED** [review]
From my re-review, I don't see any real issues, so I think it's good to be moved forward 👍

**Review by aalves08 — APPROVED** [review]
LGTM
